src/Symfony/Component/HttpKernel/Attribute/ValueResolver.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Attribute;
+
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER | \Attribute::IS_REPEATABLE)]
+class ValueResolver
+{
+    /**
+     * @param class-string<ValueResolverInterface>|string $name
+     */
+    public function __construct(
+        public string $name,
```

Comment:
What about renaming this param resolver?

Reply:
Since we want a value resolver’s name, `ValueResolver::$name` feels okay for me.

Reply:
I'm fine keeping `name` also, this reads well to me: `#[ValueResolver(name: FooResolver::class)]`

Reply:
> What about renaming this param to resolver?

While rebasing #49134, I realized that "name" is too generic and collides too easily. I've renamed to "resolver" there.